build.gradle

```diff
@@ -60,3 +61,40 @@ jacocoTestReport {
 }
 
 build.dependsOn jacocoTestReport
+
+pmd {
+    toolVersion = '5.4.2'
+    ignoreFailures = true
+    sourceSets = [sourceSets.main]
+    ruleSets = []
+    ruleSetFiles = files('pmd.xml')
+
+}
+
+pmdMain {
+    reports {
+        html.enabled = true
+        xml.enabled = true
+    }
+}
+
+build.dependsOn pmdMain
```

Comment:
minor: this line is not necessary because of the task dependency chain: `build -> check -> pmdMain`